packages/util-user-agent-node/src/index.ts

```diff
@@ -40,14 +37,17 @@ export const defaultUserAgent = ({
     sections.push([`exec-env/${env.AWS_EXECUTION_ENV}`]);
   }
 
-  const appId = await loadConfig<string | undefined>({
+  const appIdPromise = loadConfig<string | undefined>({
     environmentVariableSelector: (env) => env[UA_APP_ID_ENV_NAME],
     configFileSelector: (profile) => profile[UA_APP_ID_INI_NAME],
     default: undefined,
   })();
-  if (appId) {
-    sections.push([`app/${appId}`]);
-  }
 
-  return sections;
+  return async () => {
+    const appId = await appIdPromise;
+    if (appId) {
+      sections.push([`app/${appId}`]);
```

Comment:
Will this append `app/${appId}` multiple times, once per each time the async function is called?

It would make more sense to set appIdPromise to null after the sections was pushed, or have a boolean such as `didAppendAppId`?

Reply:
Good catch. Will fix it.